lib/net/pool.js

```diff
@@ -1126,11 +1186,16 @@ class Pool extends EventEmitter {
    * Accept an inbound socket.
    * @private
    * @param {net.Socket} socket
+   * @param {Number} type
    * @returns {Peer}
    */
 
-  createInbound(socket) {
-    const peer = Peer.fromInbound(this.options, socket);
+  createInbound(socket, type) {
+    assert(typeof type === 'number');
+    const options = new PeerOptions(this.options);
```

Comment:
Should we pass type to `PeerOptions`? Or create `StandardPeerOptions` & `BrontidePeerOptions`? Maybe this is overkill, but I don't like that a caller could miss this crucial property.